configs/components/ruby-3.2.3.rb

```diff
@@ -93,7 +93,15 @@
 
   special_flags = " --prefix=#{ruby_dir} --with-opt-dir=#{settings[:prefix]} "
 
-  if platform.name =~ /sles-15|el-8|debian-10/
+  # conditional taken from projects/_shared-compiler-settings
+  # TODO: refactor condition
+  if ((platform.is_sles? && platform.os_version.to_i >= 15) ||
+      (platform.is_el? && platform.os_version.to_i >= 8) ||
+      platform.is_debian? ||
+      (platform.is_ubuntu? && platform.os_version.to_i >= 20) ||
+      (platform.is_amazon? && platform.os_version.to_i >= 2023) ||
+      platform.is_fedora?
+     )
     special_flags += " CFLAGS='#{settings[:cflags]}' LDFLAGS='#{settings[:ldflags]}' CPPFLAGS='#{settings[:cppflags]}' "
```

Comment:
Suggest replacing the condition with:

```ruby
if settings[:supports_pie]
  special_flags += " CFLAGS='#{settings[:cflags]}' LDFLAGS='#{settings[:ldflags]}' CPPFLAGS='#{settings[:cppflags]}' "
end
```

And set `supports_pie` in the project below

Reply:
sorry wrong PR, nevermind